### PR TITLE
Unittests

### PR DESCRIPTION
**Pull request by Copro**
Please review unit-test changes.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,7 +18,12 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
   </properties>
-
+	<repositories>
```
Comment (rPraml) [inline pom.xml]: Please remove internal repository

File: pom.xml [hunk 2]
```diff
@@ -157,12 +162,18 @@
...
-      <!--<scope>test</scope>-->
-    <!--</dependency>-->
+<!-- 	<dependency>
+      <groupId>microsoft</groupId>
+      <artifactId>sqlserver-jdbc</artifactId>
+      <version>4.2</version>
+      <scope>test</scope>
+    </dependency> -->
+            <!-- Support for MS SQL Server -->
+        <dependency>
```
Comment (rPraml) [inline pom.xml]: remove this, too

File: pom.xml [hunk 3]
```diff
@@ -334,6 +345,16 @@
       </plugin>
 
     </plugins>
+<!-- mvn clean source:jar deploy -DaltDeploymentRepository=foconis::default::sftp://bamboo@repo.foconis.de/var/www/html/repo/maven-snapshot/
```
Comment (rPraml) [inline pom.xml]: and this

File: src/main/java/com/avaje/ebean/config/dbplatform/BasicMsSqlLimiter.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  @Override
+  public String limit(String dbSql, int firstRow, int maxRows) {
+
+    StringBuilder sb = new StringBuilder(50 + dbSql.length());
+    sb.append(dbSql);
+    if (!dbSql.toLowerCase().contains("order by")) {
+      sb.append(" ORDER BY 1 ");
+    }
+    //if (firstRow > 0) {
```
Comment (rPraml) [inline src/main/java/com/avaje/ebean/config/dbplatform/BasicMsSqlLimiter.java]: remove & indent

File: src/main/java/com/avaje/ebean/dbmigration/ddlgeneration/platform/MsSqlServerHistoryDdl.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  @Override
+  protected void dropTriggers(DdlBuffer buffer, String baseTable) throws IOException {
+
+    buffer.append("drop trigger ").append(updateTriggerName(baseTable)).endOfStatement();
+    buffer.append("drop trigger ").append(deleteTriggerName(baseTable)).endOfStatement();
+  }
+
+  @Override
+  protected void createTriggers(DdlWrite writer, MTable table) throws IOException {
+System.out.println("FIXME: HistorySupport is work in Progress for MsSQL");
```
Comment (rPraml) [inline src/main/java/com/avaje/ebean/dbmigration/ddlgeneration/platform/MsSqlServerHistoryDdl.java]: indent - maybe comment out all other method that creates/drops triggers as trigger syntax is completely different in MS-SQL

File: src/test/java/com/avaje/tests/batchinsert/TestBatchInsertFlush.java [hunk 6]
```diff
@@ -7,18 +7,22 @@
...
 import static org.junit.Assert.assertNotNull;
 
 public class TestBatchInsertFlush extends BaseTestCase {
-
+  
   @Test
   @Transactional(batch = PersistBatch.ALL)
   public void transactional_flushOnGetId() {
 
+    Assume.assumeFalse("Skipping test because batching not yet supported for MS SQL Server.", isMsSqlServer());
```
Comment (rPraml) [inline src/test/java/com/avaje/tests/batchinsert/TestBatchInsertFlush.java]: better use `import static org.junit.Assume.assume` - but tests should be fixed anyway.

File: src/test/java/com/avaje/tests/compositekeys/TestCore.java [hunk 7]
```diff
@@ -96,21 +107,32 @@ public void setUp() throws Exception {
...
+      testCaseTxn.rollback();
+    }
   }
 
+  @Test
   public void testFind() {
-    if (isMsSqlServer()) return;
 
-    List<Item> items = getServer().find(Item.class).findList();
+    assumeFalse("Skipping test because key feature in setUp not yet supported for MS SQL Server.", isMsSqlServer());
```
Comment (rPraml) [inline src/test/java/com/avaje/tests/compositekeys/TestCore.java]: do we have to skip the tests, too? Shouldn't an "assumeFalse" in setUp() be sufficient

File: src/test/java/com/avaje/tests/compositekeys/TestOnCascadeDeleteChildrenWithCompositeKeys.java [hunk 8]
```diff
@@ -1,6 +1,7 @@
 package com.avaje.tests.compositekeys;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assume.assumeFalse;
```
Comment (rPraml) [inline src/test/java/com/avaje/tests/compositekeys/TestOnCascadeDeleteChildrenWithCompositeKeys.java]: Yipee - all tests are passing on MsSql - so we can remove the import?

File: src/test/java/com/avaje/tests/delete/TestDeleteByIdWithPersistenceContext.java [hunk 9]
```diff
@@ -3,6 +3,7 @@
 import java.util.Arrays;
 
 import org.junit.Assert;
+import org.junit.Assume;
```
Comment (rPraml) [inline src/test/java/com/avaje/tests/delete/TestDeleteByIdWithPersistenceContext.java]: remove import

File: src/test/java/com/avaje/tests/lib/EbeanTestCase.java [hunk 10]
```diff
@@ -1,44 +0,0 @@
...
-
-import com.avaje.ebean.Ebean;
-import com.avaje.ebean.EbeanServer;
-import com.avaje.ebean.Transaction;
-
-/**
- * The base class for all Ebean test to get access to the Ebean server and do
- * some cleanup stuff after a test has run
- */
-public abstract class EbeanTestCase extends TestCase {
```
Comment (rPraml) [inline src/test/java/com/avaje/tests/lib/EbeanTestCase.java]: Yup - let's remove the old JUnit3 style "TestCase"s

File: src/test/resources/ebean.properties [hunk 11]
```diff
@@ -120,10 +120,15 @@ datasource.pg.password=unit
 datasource.pg.databaseUrl=jdbc:postgresql://127.0.0.1:5432/unit
 datasource.pg.databaseDriver=org.postgresql.Driver
 
-datasource.ms.username=test
-datasource.ms.password=test
-datasource.ms.databaseUrl=jdbc:sqlserver://192.168.1.67:1433;databaseName=test
+datasource.ms.username=zak
```
Comment (rPraml) [inline src/test/resources/ebean.properties]: please do not commit passwords ;-)

**Comment by rPraml** [comment]
Michi, that looks good :)

File: src/test/java/com/avaje/tests/lib/EbeanTestCase.java (hunk 10 above)
Comment (Copro) [inline src/test/java/com/avaje/tests/lib/EbeanTestCase.java]: Hope I did not forget to add the @Test annotation anywhere. The other issues are fixed. Also cleaned up some Checkstyle warnings where I already touched the unit-tests again.

**Pull request merged**